### PR TITLE
Merge `authentication-extension` feature branch into `master`

### PR DESCRIPTION
**Pull request by ashanthamara**
### Proposed changes in this pull request

- > $subject
- This PR will onboard following changes to the master branch which includes minor modifications to the action-mgt and action-execution components
- https://github.com/wso2/carbon-identity-framework/pull/5875
- https://github.com/wso2/carbon-identity-framework/pull/5876
- https://github.com/wso2/carbon-identity-framework/pull/5885
- https://github.com/wso2/carbon-identity-framework/pull/5904

### Related Issues
- https://github.com/wso2/product-is/issues/20748
- https://github.com/wso2/product-is/issues/20931

### CODE REVIEW

**Review by Thisara-Welmilla — DISMISSED** [review]

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 1]
```diff
@@ -2028,6 +2028,21 @@
...
+                            <Header>{{header}}</Header>
+                         {% endfor %}
+                    </ExcludedHeaders>
+                    <ExcludedParameters>
+                        {% for param in actions.types.authentication.action_request.excluded_parameters %}
+                            <Parameter>{{param}}</Parameter>
+                        {% endfor %}
+                    </ExcludedParameters>
+                </ActionRequest>
+            </Authentication>
```
Comment (malithie) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:2045]: This should be in master already

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 2]
```diff
@@ -1610,7 +1610,7 @@
     "client_assertion_type",
     "client_assertion"
   ],
-
+  "actions.types.authentication.enable": true,
```
Comment (malithie) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:1613]: same here. this should be in master already

**Review by malithie — APPROVED** [review]

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json (hunk 2 above)
Comment (ashanthamara) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:1613]: This config is for new authentication action type

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 1 above)
Comment (ashanthamara) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:2045]: This config is for new authentication action type

**Review by hwupathum — APPROVED** [review]

**Review by malithie — APPROVED** [review]

**Pull request merged**